include/fiction/technology/sidb_cluster_hierarchy.hpp

```diff
@@ -0,0 +1,1083 @@
+     */
+    template <bound_direction bound>
+    [[nodiscard]] constexpr inline double get(const uint64_t sidb_ix) const noexcept
+    {
+        return store.at(sidb_ix).at(static_cast<uint8_t>(bound));
+    }
+    /**
+     * Setter for a (partial) potential sum bound local to an SiDB.
+     *
+     * @tparam bound The potential bound to obtain.
+     * @param sidb_ix SiDB (index) to set the potential bound for.
+     * @param bound_value New bound to set.
+     */
+    template <bound_direction bound>
+    inline void set(const uint64_t sidb_ix, const double bound_value) noexcept
+    {
+        store[sidb_ix][static_cast<uint8_t>(bound)] = bound_value;
+    }
+    /**
+     * Setter for (partial) potential sum bounds local to an SiDB.
+     *
+     * @param sidb_ix SiDB (index) to set the potential bounds for.
+     * @param min New lower bound to set.
+     * @param max New upper bound to set.
+     */
+    inline void set(const uint64_t sidb_ix, const double min, const double max) noexcept
+    {
+        store[sidb_ix][static_cast<uint8_t>(bound_direction::LOWER)] = min;
+        store[sidb_ix][static_cast<uint8_t>(bound_direction::UPPER)] = max;
+    }
+    /**
+     * Relative setter for a (partial) potential sum bound local to an SiDB.
+     *
+     * @tparam bound The potential bound to update.
+     * @param sidb_ix SiDB (index) to update the potential bound of.
+     * @param bound_diff Bound difference to apply.
+     */
+    template <bound_direction bound>
+    inline void update(const uint64_t sidb_ix, const double bound_diff) noexcept
+    {
+        store[sidb_ix][static_cast<uint8_t>(bound)] += bound_diff;
+    }
+    /**
+     * Relative setter for (partial) potential sum bounds local to an SiDB.
+     *
+     * @param sidb_ix SiDB (index) to update the potential bounds of.
+     * @param min_diff Difference in lower bound potential to apply.
+     * @param max_diff Difference in upper bound potential to apply.
+     */
+    inline void update(const uint64_t sidb_ix, const double min_diff, const double max_diff) noexcept
+    {
+        store[sidb_ix][static_cast<uint8_t>(bound_direction::LOWER)] += min_diff;
+        store[sidb_ix][static_cast<uint8_t>(bound_direction::UPPER)] += max_diff;
+    }
+    /**
+     * Initialise potential bounds for the given number of SiDBs (applicable to a complete potential bounds store only).
+     *
+     * @param num_sidbs The number of SiDBs in the layout that is simulated.
+     */
+    inline void initialise_complete_potential_bounds(const uint64_t num_sidbs) noexcept
+    {
+        store.assign(num_sidbs, std::array<double, 2>{});
+    }
+
+  private:
+    /**
+     * Potential bounds are a map from SiDB indices to two values respectively representing the lower and upper bound.
+     */
+    PotentialBoundsType store{};
+};
+/**
+ * The aggregates are used in the construction; they represents information of a subhierarchy.
+ */
+#ifdef DEBUG_SIDB_CLUSTER_HIERARCHY
```

Comment:
I think we should delete this kind of macros. What do you think @marcelwa?

Reply:
Given that the `phmap` containers could be inaccurate, I think we should keep it, though under a different macro name. The highest accurracy may be delivered with STL containers.

Perhaps, even, we could introduce a global `FICTION_USE_STL_CONTAINERS` or similar and make sure the CI builds have it set. Perhaps the `operational_domain` issue would resolve then too.